### PR TITLE
feat: GET /auth/methods

### PR DESCRIPTION
**Pull request by standarius**
A prerequisite for [this task](https://app.asana.com/0/1199914526147392/1200004328990530/f).

### CODE REVIEW

**Comment by standarius** [comment]
- [ ] Some tests left to add.

File: src/Domain/UseCase/GetAuthMethods/GetAuthMethods.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      },
+    }
+  }
+
+  private getPseudoMethods(_email: string): GetAuthMethodsResponse {
+    return {
+      success: true,
+      methods: {
+        password: true,
+        // PR note: could generate a pseudo totp uuid based on the email, but won't bother and will just default to passwords for non-existent users -- unless this is insufficient?
```
Comment (karolsojko) [inline src/Domain/UseCase/GetAuthMethods/GetAuthMethods.ts:47]: should be ok - you can remove the comment and the function argument

**Review by karolsojko — APPROVED** [review]

File: src/Domain/UseCase/GetAuthMethods/GetAuthMethods.ts (hunk 1 above)
Comment (mobitar) [reply src/Domain/UseCase/GetAuthMethods/GetAuthMethods.ts:47]: I've added PR notes as GitHub comments on the line in the actual PR before in the past, seems to be a good method.

File: src/Domain/UseCase/GetAuthMethods/GetAuthMethodsResponse.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Uuid } from '../../Uuid/Uuid'
+
+export type GetAuthMethodsResponse = {
+  success: true,
+  methods: {
+    // PR note: or a password could always be implied
+    password: true,
```
Comment (mobitar) [inline src/Domain/UseCase/GetAuthMethods/GetAuthMethodsResponse.ts:7]: If this value is always going to be true then we should not have it at all. Implicit seems better.

**Pull request merged**